test/unit/trace_string_test.rb

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) SolarWinds, LLC.
+# All rights reserved.
+
+require 'minitest_helper'
+
+describe 'TraceStringTest' do
+
+  before do
+    @trace_id = rand(10 ** 32).to_s.rjust(32,'0')
+    @span_id = rand(10 ** 16).to_s.rjust(16,'0')
+    @version = "00"
+
+    @valid_tracestring =     "#{@version}-#{@trace_id}-#{@span_id}-00"
+    @sampled_tracestring =   "#{@version}-#{@trace_id}-#{@span_id}-01"
+    @broken_tracestring =    "#{@version}-f198ee56343ba864fe8b2a57d3eff7-#{@span_id}-01"
+    @malicious_tracestring = "/*TRUNCATE TABLE users*/"
+  end
+
+  it "splits a tracestring" do
+    assert AppOpticsAPM::TraceString.split(@valid_tracestring)
+    assert AppOpticsAPM::TraceString.split(@sampled_tracestring)
+    refute AppOpticsAPM::TraceString.split(@broken_tracestring)
+    refute AppOpticsAPM::TraceString.split(@malicious_tracestring)
```

Comment:
I'd refute other types too (null, number, etc.) so that test "trip wire" if something in this area of the code changes in the future.